### PR TITLE
Add `network setting` and `numa binding` recommendation 

### PR DESCRIPTION
**Pull request by tonyjohnchen**
Added a doc `PREFLIGHT.md`
Updated `maxtext_dependencies.Dockerfile` and `preflight.sh`

GCE test: https://screenshot.googleplex.com/3z74DreStvAxsdo
GKE test: https://screenshot.googleplex.com/4dVLYShbywoPj3y

### CODE REVIEW

File: PREFLIGHT.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+bash preflight.sh PLATFORM=GCE && python3 MaxText/train.py MaxText/configs/base.yml run_name=$YOUR_JOB_NAME
+```
+
+Here is an example for GKE:
+```
+bash preflight.sh PLATFORM=GKE && python3 MaxText/train.py MaxText/configs/base.yml run_name=$YOUR_JOB_NAME
+```
+
+# Optimization 2: Numa binding
+NUMA binding is recommended for enhanced performance, as it reduces memory latency and maximizes data throughput, ensuring that your high-performance applications operate more efficiently and effectively.
```
Comment (rwitten) [inline PREFLIGHT.md:19]: Can you make it clear this is v4 and v5p only?

File: README.md [hunk 2]
```diff
@@ -93,6 +93,12 @@ There are three patterns for running MaxText with more than one host.
 2. [GCE] [Running Maxtext with Multihost Jobs](getting_started/Run_MaxText_via_multihost_jobs.md) - Long Running Production Jobs with Queued Resources
 3. [GCE] [Running Maxtext with Multihost Runner](getting_started/Run_MaxText_via_multihost_runner.md) -  Fast experiments via multiple ssh connections.
 
+Once you've gotten workloads running, there are important optimizations you might want to put on your cluster.
```
Comment (rwitten) [inline README.md:96]: dupe?

**Review by rwitten — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (tonyjohnchen) [reply README.md:96]: Good catch. Thanks!

File: PREFLIGHT.md (hunk 1 above)
Comment (tonyjohnchen) [reply PREFLIGHT.md:19]: Done.

**Review by rwitten — APPROVED** [review]

**Pull request merged**